protocol-units/execution/opt-executor/src/background/transaction_pipe.rs

```diff
@@ -120,7 +121,8 @@ impl TransactionPipe {
 			self.used_sequence_number_pool.gc(epoch_ms_now);
 
 			// garbage collect the transactions in flight
-			self.transactions_in_flight.gc(epoch_ms_now);
+			let mut transactions_in_flight = self.transactions_in_flight.write().await;
+			transactions_in_flight.gc(epoch_ms_now);
```

Comment:
No need to hold this lock for the rest of the function:

```suggestion
			{
			    let mut transactions_in_flight = self.transactions_in_flight.write().await;
			    transactions_in_flight.gc(epoch_ms_now);
			}
```